zerver/webhooks/gocd/view.py

```diff
@@ -0,0 +1,49 @@
+# Webhooks for external integrations.
+
+from typing import Any, Dict, Iterable, Optional, Text
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.actions import check_send_stream_message
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_error, json_success
+from zerver.lib.validator import check_dict, check_string
+from zerver.models import UserProfile
+
+MESSAGE_TEMPLATE = (
+    u'Author: {}\n'
+    u'Build status: {} {}\n'
+    u'Details: [id]({})\n'
+    u'Comment: {}'
+)
+
+@api_key_only_webhook_view('Gocd')
+@has_request_variables
+def api_gocd_webhook(request: HttpRequest, user_profile: UserProfile,
+                     payload: Dict[str, Iterable[Dict[str, Any]]]=REQ(argument_type='body'),
+                     stream: Text=REQ(default='gocd'),
+                     topic: Optional[Text]=REQ(default='Gocd')) -> HttpResponse:
+
+    modifications = payload['build_cause']['material_revisions'][0]['modifications'][0]
+    result = payload['stages'][0]['result']
+
+    if result == "Passed":
+        emoji = ':thumbsup:'
+    elif result == "Failed":
+        emoji = ':thumbsdown:'
+    else:
+        emoji = "(No emoji specified for status '{}'.)".format(result)
+
+    body = MESSAGE_TEMPLATE.format(
+        modifications['user_name'],
+        result,
+        emoji,
+        modifications['id'],
```

Comment:
Isn't this supposed to be a link according to the message template? Seems like you are simply passing the integer ID instead of passing in a URL!